packages/components/src/breadcrumbs/test/index.tsx

```diff
@@ -128,11 +128,27 @@ describe( 'Breadcrumbs', () => {
 						</a>
 					);
 				} }
+				variant="compact"
+				showCurrentItem
 			/>
 		);
-		const secondItemLabel = THREE_ITEMS[ 1 ].label;
-		await user.click( screen.getByRole( 'link', { name: secondItemLabel } ) );
+		const firstItemLabel = FIVE_ITEMS[ 0 ].label;
+		await user.click( screen.getByRole( 'link', { name: firstItemLabel } ) );
 		expect( consoleLog ).toHaveBeenCalledTimes( 1 );
+		expect( consoleLog ).toHaveBeenCalledWith( `clicked ${ firstItemLabel }` );
+
+		// The renderProp should not be applied to the current item.
+		// We have two items with the same label due to the internal implementation
+		// for calculating when to render as 'compact'.
+		await user.click( screen.getAllByText( FIVE_ITEMS[ FIVE_ITEMS.length - 1 ].label )[ 1 ] );
```

Comment:
Nit: to avoid querying the offscreen duplicate, you can use `within( screen.getByRole( 'navigation', { name: 'Breadcrumbs' } ) )`  (as we did in a test above)

```tsx
// Since getByText would also match the offscreen copy, we are first
// finding the nav container via getByRole, which instead ignores the
// offscreen copy.
const lastItemLabel = FIVE_ITEMS[ FIVE_ITEMS.length - 1 ].label;
await user.click(
  within( screen.getByRole( 'navigation', { name: 'Breadcrumbs' } ) )
    .getByText( lastItemLabel )
);`

Reply:
Thank you! Follow up: https://github.com/Automattic/wp-calypso/pull/103382